mempool/clist_mempool.go

```diff
@@ -281,9 +281,11 @@ func (mem *CListMempool) CheckTx(tx types.Tx, cb func(*abci.Response), txInfo Tx
 			// TODO: consider punishing peer for dups,
 			// its non-trivial since invalid txs can become valid,
 			// but they can spam the same tx with little cost to them atm.
+			return ErrTxInCache
```

Comment:
why return err here?

Reply:
I was thinking that if the same peer sends the same tx then we should still return an error because that's not normal behavior (at least I think it's not normal behavior).

But I will let you be the judge for what you think is the normal behavior

Reply:
True, but inside this condition you still don't know if the same peer sent you tx again or not. For that, you need to check LoadOrStore return values (`(actual interface{}, loaded bool)`).

```suggestion
_, loaded := LoadOrStore(...)
if loaded {
			return ErrTxInCache
			} else {
			return nil
			}
```

Reply:
Right. Thanks for the heads up. I will add that now